src/skeleplex/graph/sample.py

```diff
@@ -109,3 +109,72 @@ def sample_volume_at_coordinates(
     sampled_volume = sampled_volume.reshape(*grid_shape, batch)
     # and retranspose to get batch back to the 0th dimension
     return einops.rearrange(sampled_volume, "... batch -> batch ...")
+
+
+def sample_volume_at_coordinates_lazy(
```

Comment:
I think it would be nice to have a test for this. Just something simple where the box goes outside of the image to verify that the fill value is being implemented correctly. You can use interpolation_order = 1 to make it easier.

Reply:
Added a test